d2/src/main/java/com/linkedin/d2/balancer/util/partitions/BasePartitionAccessor.java

```diff
@@ -33,5 +35,13 @@ public interface BasePartitionAccessor
    * @throws PartitionAccessException see {@link PartitionAccessException}
    */
   int getPartitionId(URI uri) throws PartitionAccessException;
+
+  /**
+   * Given the setting of the partition accessor, check if the setting can be supported
+   * @return true if supportable
+   */
+  default boolean checkSupportable(String settings) {
```

Comment:
are you thinking of adding the actual custom partition accessor in a following PR in container?

Reply:
yes